### ISSUE TITLE
Booster-bitswap Public mode causes `boostd` carshes

### ISSUE THREAD

**Issue by LexLuthr** [filecoin-project/boost#1071]
The crash is reported as below

```
2023-01-12T09:16:36.891-0500    INFO    boost-provider  storagemarket/provider.go:364   db: creating tables
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x40 pc=0x1ec8df9]

goroutine 623 [running]:
github.com/filecoin-project/boost/indexprovider.(*Wrapper).AnnounceExtendedProviders(0xc00e007400, {0x44cc0e0, 0xc0005ac040})
        /home/nonsense/boost/indexprovider/wrapper.go:143 +0x79
github.com/filecoin-project/boost/indexprovider.NewWrapper.func1.1.1()
        /home/nonsense/boost/indexprovider/wrapper.go:119 +0x25
created by github.com/filecoin-project/boost/indexprovider.NewWrapper.func1.1
        /home/nonsense/boost/indexprovider/wrapper.go:118 +0x98
```

It would seem we never set host field in the struct:
https://github.com/filecoin-project/boost/blob/f43be17448f653b63475948bd13d7f79765503b4/indexprovider/wrapper.go#L50

This is a must fix and was missed in my testing due to an unfortunate mistake on my part (systemctl was using older `boostd` binary).
We should probably re-test everything to be safe before merging the Bitswap branch.

**Comment by hannahhoward** [comment]
@LexLuthr @jacobheun I fixed in #1077. And wow, this was a really careless mistake on my part.

That said, I'd like you guys to open an issue for testing index announcements. Maybe it just means refactoring this code so you can unit test it. Maybe it means some sort of way to add a test index announcement to the docker setup (more of an integration test).

Either way, my carelessness not withstanding, this has been an issue updating this code. I haven't taken the time to do a full refactor but I think it's important.

**Comment by dirkmc** [comment]
Closed by https://github.com/filecoin-project/boost/pull/1077
